### ISSUE TITLE
[FORMATTING] CASE statements in SUM()

### ISSUE THREAD

**Issue by tanukihee** [inferrinizzard/prettier-sql#69]
**Usage**

- How are you calling / using the script?
  - I'm using https://inferrinizzard.github.io/prettier-sql/
- What SQL language(s) does this apply to?
  - The bug will not occur only if I use AWS or DB2

**Describe the bug**
Bugs occur when there are `CASE` statements in `SUM()`.

1. When there is only one `SUM(CASE ...)`, like
```sql
select sum(case a when foo then bar end) from quaz
```
**Expected Output**
```sql
SELECT
  SUM(CASE a
  WHEN foo
  THEN bar END
)
FROM
  quaz
```

**Actual Output**
```sql
SELECT
  SUM(CASE a
  WHEN foo
  THEN barEND
)
FROM
  quaz
```

Note there should be a space before the `END` .

2. When there is something else after `SUM(CASE ...)`, like
```sql
select sum(case a when foo then bar end), b from quaz
```
**Expected Output**
```sql
SELECT
  SUM(CASE a
  WHEN foo
  THEN bar END
  )
  , b
FROM
  quaz
```

**Actual Output**
```shell
An Error Occurred, please report this at:
https://github.com/inferrinizzard/prettier-sql/issues

Stack Trace:
TypeError: Cannot read properties of null (reading '1')
    at https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:9589
    at Array.map (<anonymous>)
    at https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:9539
    at I.value (https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:9715)
    at I.value (https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:8987)
    at I.value (https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:8800)
    at Module.VT (https://unpkg.com/prettier-sql@latest/dist/prettier-sql.min.js:1:184856)
    at HTMLTextAreaElement.format (https://inferrinizzard.github.io/prettier-sql/index.js:42:31)

```

**Comment by tanukihee** [comment]
Maybe the `bug` label is more suitable😂

**Comment by CreationEdge** [comment]
This occurs for BigQuery as well.

But I was able to circumvent it by setting my Line Width to 40 (the issue occurs at 60 and 80). 

Additionally, it doesn't occur consistently. If I have 8 such `SUM(CASE ... END)` statements in a row, it may only be the last 3 that get the incorrect formatting. In may case, turning `NULL END` into `NULLEND`.

**Comment by inferrinizzard** [comment]
I can't seem to reproduce the final error you reported but there is a fix out for the END spacing issue in #79

**Comment by tanukihee** [comment]
> I can't seem to reproduce the final error you reported but there is a fix out for the END spacing issue in #79 

My fault. Please change the `CommaPosition` option to `before`.

![Screenshot_2022-04-05-12-37-07-300_com.microsoft.emmx.jpg](https://user-images.githubusercontent.com/34062491/161679803-69aa7ca4-654b-462c-98f6-abe9b2bf17c7.jpg)
